src/predict.rs

```diff
@@ -422,6 +460,21 @@ where
   fn pred_smooth_h(
     output: &mut [T], stride: usize, above: &[T], left: &[T]
   ) {
+    #[cfg(all(target_arch = "x86_64", not(windows)))]
+    {
+      if size_of::<T>() == 1 && is_x86_feature_detected!("avx2") {
+        return unsafe {
+          rav1e_ipred_smooth_h_avx2(
```

Comment:
Is there a way to shorten this given that most of the ipred functions have the same call signature?

Reply:
We're planning to unify the signatures and/or rework the dispatch in the near future.
In the first case, this can be a single macro.